### ISSUE TITLE
[BUG] Login of the user doesn't stay if I refresh the page ( Non - consistent )

### ISSUE THREAD

**Issue by krishnaacharyaa** [krishnaacharyaa/wanderlust#172]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I login and then refresh the page, it immediately shows me that I am logged out and asks me to login in again, where as according to the logic it should have waited for 4mins and only after 4mins it should have showed me to login again

### Expected Behavior

Until it has hit four minutes I shouldn't be logged out after refresh

### Reproduction steps

```bash
1. Go to https://wanderlust-beta.vercel.app/
2. Register/Login 
3. See the create-post and logout button
4. Refresh the page.
5. You'll again see the login button
```

### Screenshots

Login and after Refresh

![image](https://github.com/krishnaacharyaa/wanderlust/assets/116620586/9f027936-38c0-4c3a-b15a-ff67b2bd8d45)

**Comment by krishnaacharyaa** [comment]
cc: @0xmohitsen

**Comment by krishnaacharyaa** [comment]
@Janith-Chamikara can you kindly look into this...

**Comment by Ameerjafar** [comment]
the cookie is not set in the cookie section when we logged in the website and i am currently working on it sir.

**Comment by krishnaacharyaa** [comment]
Sure Sir @Ameerjafar

**Comment by Ameerjafar** [comment]
can set the cookie in the frontend sir @krishnaacharyaa

**Comment by Ameerjafar** [comment]
hello sir I need your help @krishnaacharyaa

**Comment by Ameerjafar** [comment]
sir can you check whether token set in the cookie section or not please let me know sir.

**Comment by krishnaacharyaa** [comment]
Hi @Ameerjafar please try figuring out, and ask in the team's discord for help, if it doesn't work at all, worst case. We can sync up.

p.s: You can find it in the console of the web browser in the application section there you'll have cookies section... Kindly check there

**Comment by Ameerjafar** [comment]
@krishnaacharyaa I found the exact error sir. whenever we are refreshing the page the home page(header-layout) component will re-render if it is re-render the usestate is set to null because it is not persistent when the page is refresh even it is static.

**Comment by krishnaacharyaa** [comment]
@Ameerjafar Thats a great insight, precisely it is because we are storing the token in the state, instead we should store it in the cookies, and revalidate cookies everytime, if in state, it loses persistence

**Comment by Ameerjafar** [comment]
can i make pr for this i solved this issue sir

**Comment by krishnaacharyaa** [comment]
Yes please @Ameerjafar , please make sure you followed all the contributing guidelines 
https://github.com/krishnaacharyaa/wanderlust/blob/9b11b769bb23150b746296cf9008056633d21921/.github/CONTRIBUTING.md#guidelines-for-contributions

**Comment by thefarhanahmad** [comment]
Hi there! I'm excited to contribute to this project and would love to work on this issue. Could you please assign it to me? Thanks!

**Comment by Ameerjafar** [comment]
@thefarhanahmad i have completed this issue just I need to make the pr according to the guidelines has given.

**Comment by 0xmohitsen** [comment]
> @thefarhanahmad i have completed this issue just I need to make the pr according to the guidelines has given.

Yes, follow the guidelines.

**Comment by Ameerjafar** [comment]
I have made the pr can you check that sir @krishnaacharyaa

**Comment by thefarhanahmad** [comment]
Ok, I will check.

**Comment by Ameerjafar** [comment]
@krishnaacharyaa can i make pr now according to the guildlines you have provided sir.

**Comment by krishnaacharyaa** [comment]
Sure @Ameerjafar , But I see we are still not very sure on what is the root cause , and how to gracefully handle it from ground up, let's come to a root cause and approach and then raise PR based on the guidelines. That would be much better in my opinion and both of our time would be saved :)

**Comment by Ameerjafar** [comment]
@krishnaacharyaa i have raised the pr for this error sir.

**Comment by krishnaacharyaa** [comment]
@Ameerjafar reviewed it :)

**Comment by Ankit389** [comment]
Hello Sir,

i can fix this  issue sir @krishnaacharyaa

Description

Then, in your login controller, after successful authentication, you can generate a token and send it back to the client:

const { generateToken } = require('../utils/auth');

const loginUser = async (req, res) => {
    // Assuming successful authentication and getting user data
    const userData = { userId: 123, username: 'example_user' };
    
    // Generate JWT token
    const token = generateToken(userData);

    res.json({ token });
};


GSSOC'24
